astropy/units/format/cds.py

```diff
@@ -46,6 +45,8 @@ class CDS(Base):
     _times: ClassVar[str] = "x"
     _scale_unit_separator: ClassVar[str] = ""
 
+    _deprecated_units: ClassVar[dict[str, UnitBase]] = {}
```

Comment:
I was wondering why this was necessary, so had a bit of a look. The alternative would seem to be to rename the current `_units` to `_all_units` or `_generate_unit_names` and let it `return names, set()` (like fits does in `_generate_unit_names()`) - that keeps the implementation more consistent with that of generic, fits, and ogip.

Independently, it looks like it should be an empty set, not dict.

Reply:
Indeed, partially why I suggested to rename `_units` to `_all_units` first (and change the return statement). That keeps the implementation as similar as possible, and (I think) makes it even more obvious that the whole deprecated units bit should be moved to OGIP.